Jim-Hodapp-Coaching/refactor-platform-fe
# Handle setting and updating an overarching goal

## Description
This PR sets and updates an overarching goal for this coaching session.

**Note:** pairs with [this backend PR](https://github.com/Jim-Hodapp-Coaching/refactor-platform-rs/pull/69)

#### GitHub Issue: None

### Changes
* Adds a new OverarchingGoalContainer component that contains the OverarchingGoal, the AgreementList and the ActionList components
* Adds a new OverarchingGoal component that handles setting/updating an overarching goal

### Screenshots / Videos Showing UI Changes (if applicable)

Drawer closed:
![Jim __ Caleb _ Oct 10_ 2024](https://github.com/user-attachments/assets/6b46e432-0112-4ca2-afdc-30833616463f)

Drawer opened:
![Jim __ Caleb _ Oct 10_ 2024](https://github.com/user-attachments/assets/ad8f637b-ab5b-4411-910c-cc75c2d55337)

### Testing Strategy
1. Navigate to a coaching session page
2. Click the down chevron button to expand the drawer
3. You should be able to click in the overarching goal line into a textbox and type a new goal
4. Either click "Set" or press <Enter> with the focus in the textbox
5. Notice that the drawer closes and the overarching goal label should be updated with whatever you entered.


### Concerns
None
